_posts/2020-3-27-SciML.md

```diff
@@ -0,0 +1,380 @@
+# pip install diffeqpy
+from diffeqpy import de
+# diffeqpy.install()
+
+def f(du,u,p,t):
+    x, y, z = u
+    sigma, rho, beta = p
+    du[0] = sigma * (y - x)
+    du[1] = x * (rho - z) - y
+    du[2] = x * y - beta * z
+
+def g(du,u,p,t):
+    du[0] = 0.3*u[0]
+    du[1] = 0.3*u[1]
+    du[2] = 0.3*u[2]
+
+numba_f = numba.jit(f)
+numba_g = numba.jit(g)
+u0 = [1.0,0.0,0.0]
+tspan = (0., 100.)
+p = [10.0,28.0,2.66]
+prob = de.SDEProblem(numba_f, numba_g, u0, tspan, p)
+sol = de.solve(prob)
+
+# Now let's draw a phase plot
+
+ut = numpy.transpose(sol.u)
+from mpl_toolkits.mplot3d import Axes3D
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+ax.plot(ut[0,:],ut[1,:],ut[2,:])
+plt.show()
+```
+
+### We provide tools for researching methods in scientific machine learning
+
+Last but not least, we support the research activities of practitioners in
+scientific machine learning. Tools like [DiffEqDevTools.jl](https://github.com/SciML/DiffEqDevTools.jl)
+and [RootedTrees.jl](https://github.com/SciML/RootedTrees.jl) make it easy to
+create and benchmark new methods and accelerate the publication
+process for numerical researchers. Our wrappers for external tools like
+[FEniCS.jl](https://github.com/SciML/FEniCS.jl) and
+[SciPyDiffEq.jl](https://github.com/SciML/SciPyDiffEq.jl) make it easy to perform
+cross-platform comparisons. Our stack is entirely written within Julia,
+which means every piece can be tweaked on the fly, making it easy to mix
+and match Hamiltonian integrators with neural networks to discover new scientific
+applications. Our issues and [chat channel](https://gitter.im/JuliaDiffEq/Lobby)
+serve as places to not just debug existing software, but also discuss new
+methods and help create high performance implementations.
+
+In addition, we support many student activities to bring new researchers into the
+community. Many of the maintainers of our packages, like Yingbo Ma, Vaibhav Dixit,
+Kanav Gupta, Kirill Zubov, etc. all started as one of our over 50 student developers
+from past [Google Summer of Code](https://summerofcode.withgoogle.com/) and other
+[Julia Seasons of Contributions](https://julialang.org/jsoc/).
+
+## The Goal of SciML
+
+When you read a paper that is
+[mixing neural networks with differential equations](https://arxiv.org/abs/2001.04385)
+or [designing new neural networks that satisfy incompressibility for modeling Navier-Stokes](https://arxiv.org/abs/1903.00033),
+you should be able to go online and find tweakable, high quality, and highly
+maintained package implementations of these methodologies to either start using
+for your scientific research, or utilize as a starting point for furthering the
+methods of scientific machine learning. For this reason, the goal of the SciML
+OSS organization is to be a hub for the development of robust cross-language
+scientific machine learning software. **In order to make this a reality, we as
+an organization have to commit to the following principles**:

```

Comment:
```suggestion
an organization commit to the following principles**:
```

Is this a statement of fact for libraries which can be considered part of SciML, or a call to action?

Reply:
It's somewhat a statement of fact, but really defining what we've had as a working principle in our head